R/misc.R

```diff
@@ -436,13 +436,36 @@ check_type <- function(dat, quant = TRUE, types = NULL, call = caller_env()) {
   }
 
   if (!all(all_good)) {
-    rlang::abort(
-      paste0(
-        "All columns selected for the step",
-        " should be ",
-        label,
-        "."
-      ),
+    info <- get_types(dat)[!all_good, ]
+    classes <- map_chr(info$type, function(x) x[1])
+    counts <- vctrs::vec_split(info$variable, classes)
+    counts$count <- lengths(counts$val)
+    counts$text_len <- cli::console_width() - 19 - (counts$count > 1) -
+      nchar(counts$key)
+
+    # cli::ansi_collapse() doesn't work for length(x) == 1
+    # https://github.com/r-lib/cli/issues/590
+    variable_collapse <- function(x, width) {
+      x <- paste0("{.var ", x, "}")
+      if (length(x) == 1) {
+        res <- cli::ansi_strtrim(x, width = width)
+      } else {
+        res <- cli::ansi_collapse(x, width = width, style = "head")
+      }
+      res
+    }
+
+    problems <- glue::glue_data(
+      counts,
+      "{count} {key} variable{ifelse(count == 1, '', 's')} \\
+      found: {purrr::map2_chr(val, text_len, variable_collapse)}"
+    )
+    names(problems) <- rep("*", length(problems))
+
+    message <- "All columns selected for the step should be {label}."
```

Comment:
```suggestion
    message <- "All columns selected for the step should be {.or {types}}."
```

Some of the relevant lines are outside of where I'm allowed to comment/suggest, but I noticed that there's an additional comma added when there are only two options in `glue::glue_collapse(types, sep = ", ", last = ", or ")`. e.g., from the snaps for `step_cut()`:

> `x All columns selected for the step should be double, or integer.`

cli has a `.or` inline markup option that would come in handy here.

Besides this, these are all that would need to change, I think:

<img width="697" alt="Screenshot 2023-10-11 at 10 26 24 AM" src="https://github.com/tidymodels/recipes/assets/35748691/a08724e5-e025-40ea-956f-5e083eb5dd8d">

Reply:
closed in [`d962daa` (#1235)](https://github.com/tidymodels/recipes/pull/1235/commits/d962daaa49a3c001e6b423412ff80d40970d1d86)